include/zephyr/toolchain/gcc.h

```diff
@@ -244,6 +244,10 @@ do {                                                                    \
 #endif
 #endif
 
+#ifndef __malloc_like_with_free
```

Comment:
It's probably better to create zephyr macros instead of attempting to re-use those from picolibc. That way you wouldn't need `#ifndef` here, and if picolibc changed its macro in the future, zephyr would continue to work.  Identifiers with a leading underscore are supposed to be reserved, so Zephyr shouldn't be defining them anyways. `z_malloc_like_with_free`? I dunno; names are hard.

Reply:
I'm not good at naming. I was using the `__used()` or `__printf_like()` pattern we have in `gcc.h`. I'm sure these are from Linux. There aren't many defines with the `z_` prefix in `gcc.h` either, but I understand that leading underscores should be reserved.

Does anyone have any suggestions?

Reply:
My only strong suggestion is to use something other than the name defined by picolibc.

Reply:
I've changed to `__deallocate_with()` in the last push.